agent/connect/ca/provider_vault.go

```diff
@@ -344,18 +344,7 @@ func (v *VaultProvider) GenerateCAChain() (CAChainResult, error) {
 		rootChain = rootPEM
 	}
 
-	intermediate, err := v.ActiveLeafSigningCert()
-	if err != nil {
-		return CAChainResult{}, fmt.Errorf("error fetching active intermediate: %w", err)
-	}
-	if intermediate == "" {
-		intermediate, err = v.GenerateLeafSigningCert()
-		if err != nil {
-			return CAChainResult{}, fmt.Errorf("error generating intermediate: %w", err)
-		}
-	}
-
-	return CAChainResult{PEM: rootChain, IntermediatePEM: intermediate}, nil
```

Comment:
I had initially bundled this here because I had considered the "CA chain" to be both root CA and optional intermediate CAs. While fixing the bug I concluded that it was simpler to leave `GenerateCAChain` as "root CA only" while treating intermediate CA's separately when applicable.